toolchain/parse/typed_nodes.h

```diff
@@ -1032,6 +1034,51 @@ struct DesignatorExpr {
   NodeIdOneOf<IdentifierName, SelfTypeName> name;
 };
 
+// FIXME: RequirementRewrite?
+struct RequirementAssign {
+  static constexpr auto Kind = NodeKind::RequirementAssign.Define(
+      {.category = NodeCategory::Requirement, .child_count = 2});
+  AnyExprId lhs;
+  Lex::EqualTokenIndex token;
+  AnyExprId rhs;
+};
+
+// FIXME: RequirementEqualEqual?
+struct RequirementEquals {
+  static constexpr auto Kind = NodeKind::RequirementEquals.Define(
+      {.category = NodeCategory::Requirement, .child_count = 2});
+  AnyExprId lhs;
+  Lex::EqualEqualTokenIndex token;
+  AnyExprId rhs;
+};
+
+struct RequirementImpls {
+  static constexpr auto Kind = NodeKind::RequirementImpls.Define(
+      {.category = NodeCategory::Requirement, .child_count = 2});
+  AnyExprId lhs;
+  Lex::ImplsTokenIndex token;
+  AnyExprId rhs;
+};
+
+// virtual node
+// FIXME: WhereType? WhereOperand?
```

Comment:
`WhereOperand` would be consistent with what we do for `and` / `or`.

Reply:
Done.